src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowMonitor.scala

```diff
@@ -121,4 +124,12 @@ class WorkflowMonitor(parent: ActorRef,
   private def getWorkspaceContext(workspaceName: WorkspaceName, txn: RawlsTransaction): WorkspaceContext = {
     containerDAO.workspaceDAO.loadContext(workspaceName, txn).getOrElse(throw new RawlsException(s"workspace ${workspaceName} does not exist"))
   }
+
+  private def getServiceAccountUserInfo = {
+    val expiresInSeconds: Long = Option(serviceAccountCred.getExpiresInSeconds).map(_.toLong).getOrElse(0)
```

Comment:
this will refresh the token if there are less than 10 seconds remaining on it, and then it's used to check workflow status, is 10 seconds enough?